engine/crates/engine-v2/src/engine.rs

```diff
@@ -57,6 +58,7 @@ pub struct Engine<R: Runtime> {
     pub(crate) runtime: R,
     operation_metrics: GraphqlOperationMetrics,
     auth: AuthService,
+    retry_budgets: Vec<Option<RetryBudget>>,
```

Comment:
We need this here and not in config since it's state, and it can't be serialized.